src/mame/rm/rm380z.cpp

```diff
@@ -268,30 +268,28 @@ void rm380z_state::configure(machine_config &config)
 	RAM(config, RAM_TAG).set_default_size("56K");
 
 	/* floppy disk */
-	FD1771(config, m_fdc, 16_MHz_XTAL / 16);
-
-	FLOPPY_CONNECTOR(config, m_floppy0, rm380z_floppies, "mds", floppy_image_device::default_mfm_floppy_formats).set_fixed(true);
-	FLOPPY_CONNECTOR(config, m_floppy1, rm380z_floppies, "mds", floppy_image_device::default_mfm_floppy_formats).set_fixed(true);
+	if (bFDS)
```

Comment:
A better pattern for this would be to have a base_configure function that sets up all of the non-optional hardware and then call it from the specific machine's configure functions and add on the rest.

Look at https://github.com/mamedev/mame/blob/master/src/mame/apple/maclc3.cpp where I do exactly this: maclc3_base() configures the non-optional hardware and the actual machines call it and then do their additional things.